### PR TITLE
making script compatible with new changelog format

### PR DESCRIPTION
**Pull request by bsipocz**
This is a hacky way to make this work with the new changelog format. Probably there is a nicer way to do it, I rather dislike the double if...

### CODE REVIEW

File: changebot/changelog.py [hunk 1]
```diff
@@ -25,8 +25,12 @@ def find_prs_in_changelog_by_section(content):
     subcontent = ''
     previous = None
 
+    new_changelog_format = False
+    
     for line in content.splitlines():
-        if '-------' in line:
+        if '=======' in line:
+            new_changelog_format = True
+        if '=======' in line or (not new_changelog_format and '-------' in line):
```
Comment (pllim) [inline changebot/changelog.py:33]: Well... It looks slightly less repetitive if you replace `'=======' in line` here with `new_changelog_format`. I don't see any easy solution to get rid of the double-if without refactoring the rest below (and even then might still look ugly).

Comment (bsipocz) [reply changebot/changelog.py:33]: But then the expression will always be True, not just for the lines that has "====" in them. Ideally the first if need to run only until the first occurrence of "====", from there on we know it's a new changelog.

Comment (pllim) [reply changebot/changelog.py:33]: Not sure if I understand (it *is* past my bedtime...), but I don't see any difference logically between
```python
if '=======' in line or (not new_changelog_format and '-------' in line)
```
and
```python
if new_changelog_format or (not new_changelog_format and '-------' in line)
```

... anyway feel free to ignore if I misread your code.

Comment (bsipocz) [reply changebot/changelog.py:33]: Once ``new_changelog_format`` is set True, it stays True always. (but it also passed my bedtime, long time ago :) )

**Comment by astrofrog** [comment]
I'll merge this for now as I think it should work. I have a fix in the works to properly parse the file via an rst parser, but this works for now.

**Pull request merged**